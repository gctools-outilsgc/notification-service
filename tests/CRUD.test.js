const fs = require("fs");
const mutations = require("../src/resolvers/Mutations");
const query = require("../src/resolvers/Query");
const { cleanUp, setPrisma } = require("./init/prismaTestInstance");
const parent = {};
const ctx = {
  prisma: setPrisma()
};

afterAll(async (done) => {
    await cleanUp(ctx);
    done();
});

test("create notification", async() => {
  const args = {
    gcID: "sde432sde",
    appID: "154879",
    actionLevel: "Featured",
    email: {
      from: "from@test.ca",
      to: "to@test.ca",
      subject: "Test email subject",
      body: "Test email body",
      status: "Sent"
    },
    online: {
      titleEn: "Test online title English",
      titleFr: "Test titre français en ligne",
      descriptionEn: "Test online description English",
      descriptionFr: "Test description français en ligne",
      viewed: false
    }
  };

  const info = "{ gcID, appID, actionLevel, email { from, subject, body, status }, online { titleEn, titleFr, descriptionEn, descriptionFr, viewed } }";

  expect(
    await mutations.createNotification(parent, args, ctx, info),
  ).toMatchSnapshot();

});

test("Modify Notification", async() => {
    var notificationID = await query.notifications(parent, {gcID:"sde432sde"}, ctx, "{id}");
    var args = {
        id: notificationID[0].id,
        online: {
                titleEn:"My test edited",
                titleFr:"Mon test mondifié",
                descriptionEn:"My description edited",
                descriptionFr:"Ma description modifié",
                viewed:true
        }
    };

    const info = "{ online{titleEn,titleFr,descriptionEn,descriptionFr,viewed}}";

    expect(
        await mutations.updateNotification(parent, args, ctx, info)
    ).toMatchSnapshot();

});

test("Query all notifications", async() => {
    const info = "{gcID, appID, actionLevel, actionLink, email{ from, to, subject, body, status, html, sendError}, online{ titleEn, titleFr, descriptionEn, descriptionFr, viewed}, whoDunIt{ gcID, teamID, organizationID}}";
    
    expect(
      await query.notifications(parent, {}, ctx, info),
    ).toMatchSnapshot();
  
  });
  