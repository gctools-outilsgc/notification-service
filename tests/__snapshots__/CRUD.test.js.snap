// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modify Notification 1`] = `
Object {
  "online": Object {
    "descriptionEn": "My description edited",
    "descriptionFr": "Ma description modifié",
    "titleEn": "My test edited",
    "titleFr": "Mon test mondifié",
    "viewed": true,
  },
}
`;

exports[`Query all notifications 1`] = `
Array [
  Object {
    "actionLevel": "Featured",
    "actionLink": null,
    "appID": "154879",
    "email": Object {
      "body": "Test email body",
      "from": "from@test.ca",
      "html": false,
      "sendError": null,
      "status": "Sent",
      "subject": "Test email subject",
      "to": "to@test.ca",
    },
    "gcID": "sde432sde",
    "online": Object {
      "descriptionEn": "My description edited",
      "descriptionFr": "Ma description modifié",
      "titleEn": "My test edited",
      "titleFr": "Mon test mondifié",
      "viewed": true,
    },
    "whoDunIt": null,
  },
]
`;

exports[`create notification 1`] = `
Object {
  "actionLevel": "Featured",
  "appID": "154879",
  "email": Object {
    "body": "Test email body",
    "from": "from@test.ca",
    "status": "Sent",
    "subject": "Test email subject",
  },
  "gcID": "sde432sde",
  "online": Object {
    "descriptionEn": "Test online description English",
    "descriptionFr": "Test description français en ligne",
    "titleEn": "Test online title English",
    "titleFr": "Test titre français en ligne",
    "viewed": false,
  },
}
`;
